### PR TITLE
Bug 1738421: Re-create default nodePolicy CR when it deleted

### PR DESCRIPTION
**Pull request by pliurh**
Also let operator watch secondary resources, so that secondary resondary resource can be restored.

Fix https://bugzilla.redhat.com/show_bug.cgi?id=1738421

### CODE REVIEW

File: pkg/controller/sriovnetworknodepolicy/sriovnetworknodepolicy_controller.go [hunk 1]
```diff
@@ -106,6 +176,22 @@ func (r *ReconcileSriovNetworkNodePolicy) Reconcile(request reconcile.Request) (
 	defaultPolicy := &sriovnetworkv1.SriovNetworkNodePolicy{}
 	err := r.client.Get(context.TODO(), types.NamespacedName{Name: DEFAULT_POLICY_NAME, Namespace: Namespace}, defaultPolicy)
 	if err != nil {
+		if errors.IsNotFound(err) {
+			// Default policy object not found, create it.
+			defaultPolicy.SetNamespace(Namespace)
```
Comment (zshi-redhat) [inline pkg/controller/sriovnetworknodepolicy/sriovnetworknodepolicy_controller.go]: Is it possible that namespace from reconcile.request is not from `sriov-network-operator`?

Comment (pliurh) [reply pkg/controller/sriovnetworknodepolicy/sriovnetworknodepolicy_controller.go]: Currently, the operator is designed to only work in namespace `sriov-network-operator`.

**Comment by pliurh** [comment]
@zshi-redhat fixed

**Pull request merged**